src/Microsoft.IdentityModel.Logging/LoggerContext.cs

```diff
@@ -0,0 +1,81 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Microsoft.IdentityModel.Logging
+{
+    /// <summary>
+    /// An opaque context used to store work when working with authentication artifacts.
+    /// </summary>
+    public class LoggerContext
+    {
+        /// <summary>
+        /// Instantiates a new <see cref="LoggerContext"/> with a default activityId.
+        /// </summary>
+        public LoggerContext()
+        {
+        }
+
+        /// <summary>
+        /// Instantiates a new <see cref="LoggerContext"/> with an activityId.
+        /// </summary>
+        /// <param name="activityId"></param>
+        public LoggerContext(Guid activityId)
+        {
+            ActivityId = activityId;
+        }
+
+        /// <summary>
+        /// Gets or sets a string that helps with setting breakpoints when debugging.
+        /// </summary>
+        public string Id { get; set; } = string.Empty;
```

Comment:
public string Id will replace an 'internal' property with the same name in SAL's S2SContext class. In the next SAL PR I'll make sure to remove the internal property to avoid compiler getting confused.

Reply:
Why are we exposing this property if it's used only internally?

Reply:
@GeoK It is internal in SAL, but we would like it to be public to aid in debugging.